synapse/federation/sender/__init__.py

```diff
@@ -393,34 +448,13 @@ async def handle_room_events(events: Iterable[EventBase]) -> None:
         finally:
             self._is_processing = False
 
-    async def _send_pdu(self, pdu: EventBase, destinations: Iterable[str]) -> None:
-        # We loop through all destinations to see whether we already have
-        # a transaction in progress. If we do, stick it in the pending_pdus
-        # table and we'll get back to it later.
-
-        destinations = set(destinations)
-        destinations.discard(self.server_name)
-        logger.debug("Sending to: %s", str(destinations))
+    def _distribute_pdus(self, destination_to_pdus: Dict[str, List[EventBase]]) -> None:
+        for destination, pdus in destination_to_pdus.items():
+            logger.debug("Sending %d to: %s", len(pdus), destination)
 
-        if not destinations:
-            return
-
-        sent_pdus_destination_dist_total.inc(len(destinations))
-        sent_pdus_destination_dist_count.inc()
```

Comment:
These became
```py
sent_pdus_destination_dist_total.inc(len(pdus))
```
per every destination.

I'm fairly sure this is incorrect, what could I replace this with?

Reply:
Then imma guess that the new approach is correct? Then i don't understand what the old counter was doing with an additional `.inc` after having done so for every destination

Reply:
note that there are two separate counters